src/query/src/promql/planner.rs

```diff
@@ -1385,7 +1400,32 @@ impl PromPlanner {
             .collect();
         self.ctx.tag_columns = tags;
 
-        Ok(())
+        Ok(None)
+    }
+
+    /// Setup [PromPlannerContext]'s state fields for a non existent table
+    /// without any rows.
+    fn setup_context_for_empty_metric(&mut self) -> Result<LogicalPlan> {
+        self.ctx.time_index_column = Some(SPECIAL_TIME_FUNCTION.to_string());
+        self.ctx.reset_table_name_and_schema();
+        self.ctx.tag_columns = vec![];
+        self.ctx.field_columns = vec![DEFAULT_FIELD_COLUMN.to_string()];
+
+        // The table doesn't have any data, so we set start to 0 and end to -1.
+        let plan = LogicalPlan::Extension(Extension {
+            node: Arc::new(
+                EmptyMetric::new(
+                    0,
+                    -1,
+                    self.ctx.interval,
+                    SPECIAL_TIME_FUNCTION.to_string(),
+                    DEFAULT_FIELD_COLUMN.to_string(),
```

Comment:
not sure if we should use `GREPTIME_TIMESTAMP` and `GREPTIME_VALUE` instead. Both are Ok to me

Reply:
Most places in the promql planner use these names, so I follow their patterns.